core/drivers/crypto/se050/adaptors/apis/apdu.c

```diff
@@ -524,3 +524,205 @@ sss_status_t se050_scp03_send_rotate_cmd(pSe05xSession_t ctx,
 
 	return kStatus_SSS_Fail;
 }
+
+sss_status_t se050_ecc_gen_shared_secret(pSe05xSession_t ctx, uint32_t kid,
+					 struct se050_ecc_keypub *keypub,
+					 uint8_t *secret, unsigned long *len)
+{
+	smStatus_t status = SM_OK;
+	uint8_t *pubkey = NULL;
+	uint8_t *buf = NULL;
+	size_t pubkey_len = 0;
+
+	if (!keypub || !secret || !len)
+		return kStatus_SSS_Fail;
+
+	pubkey_len = keypub->x_len + keypub->y_len + 1;
```

Comment:
The lines 540-548 are repeated in two functions below also, perhaps a small helper function would be in order.

Reply:
ok...would doing something like this be acceptable (just so I can reuse the same static function later on)


```
struct se050_ecc_keypair {
	struct se050_ecc_keypub pub;
	uint8_t *d;
	size_t d_len;
};
```

Reply:
Nice